src/solver/variable/include/antares/solver/variable/economy/minOfDispatchableGen.h

```diff
@@ -0,0 +1,319 @@
+            {
+                pValuesForTheCurrentYear[numSpace].resize(nbColumns_);
+            }
+
+            for (unsigned int numSpace = 0; numSpace < pNbYearsParallel; numSpace++)
+            {
+                for (unsigned int i = 0; i != nbColumns_; ++i)
+                {
+                    pValuesForTheCurrentYear[numSpace][i].initializeFromStudy(*study);
+                }
+            }
+
+            for (unsigned int i = 0; i != nbColumns_; ++i)
+            {
+                AncestorType::pResults[i].initializeFromStudy(*study);
+                AncestorType::pResults[i].reset();
+            }
+        }
+        else
+        {
+            AncestorType::pResults.clear();
+        }
+        // Next
+        NextType::initializeFromArea(study, area);
+    }
+
+    size_t getMaxNumberColumns() const
+    {
+        return nbColumns_ * ResultsType::count;
+    }
+
+    void yearBegin(unsigned int year, unsigned int numSpace)
+    {
+        // Reset the values for the current year
+        for (unsigned int i = 0; i != nbColumns_; ++i)
+        {
+            pValuesForTheCurrentYear[numSpace][i].reset();
+        }
+        // Next variable
+        NextType::yearBegin(year, numSpace);
+    }
+
+    void yearEnd(unsigned int year, unsigned int numSpace)
+    {
+        for (unsigned int column = 0; column < nbColumns_; column++)
+        {
+            pValuesForTheCurrentYear[numSpace][column].computeStatisticsForTheCurrentYear();
+        }
+
+        // Next variable
+        NextType::yearEnd(year, numSpace);
+    }
+
+    void computeSummary(std::map<unsigned int, unsigned int>& numSpaceToYear,
+                        unsigned int nbYearsForCurrentSummary)
+    {
+        for (unsigned int numSpace = 0; numSpace < nbYearsForCurrentSummary; ++numSpace)
+        {
+            VariableAccessorType::ComputeSummary(pValuesForTheCurrentYear[numSpace],
+                                                 AncestorType::pResults,
+                                                 numSpaceToYear[numSpace]);
+        }
+        // Next variable
+        NextType::computeSummary(numSpaceToYear, nbYearsForCurrentSummary);
+    }
+
+    void hourBegin(unsigned int hourInTheYear)
+    {
+        // Next variable
+        NextType::hourBegin(hourInTheYear);
+    }
+
+    void hourForEachArea(State& state, unsigned int numSpace)
+    {
+        auto area = state.area;
+        auto& thermal = state.thermal;
+        for (auto& cluster: area->thermal.list.each_has_min_gen())
+        {
+            unsigned int groupNumber = groupToNumbers_[cluster->getGroup()];
+            double minGen = cluster->PthetaInf[state.hourInTheYear];
+            double production = thermal[area->index]
+                                  .thermalClustersProductions[cluster->enabledIndex];
+
+            pValuesForTheCurrentYear[numSpace][groupNumber][state.hourInTheYear] += std::min(
+              production,
+              minGen);
```

Comment:
Not sure I understand this, isn't this always the case that 

production >= minGen

?

Reply:
- **minGen** is made from only : 
   - **min gen modulation** (which is a TS of doubles inside [0, 1]) 
   -  **nominal capacity**.
- **available capacity** is a TS on its own.

If, for hours in the week, **available capacity** (which is an upper bound for production) decreases under **minGen**,  then the lower bound for production at these hours is **available capacity**. So, for these hours, production is exactly **available capacity**.

Indeed, we have : 
```c++
Pt.PuissanceMinDuPalierThermique[hourInWeek] = 
    (Pt.PuissanceDisponibleDuPalierThermique[hourInWeek] < cluster->PthetaInf[hourInYear]) ?
        Pt.PuissanceDisponibleDuPalierThermique[hourInWeek] : 
        cluster->PthetaInf[hourInYear];
```